ShaderGenerators/UnityShaderParser.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System.Text.RegularExpressions;
+
+namespace SourceGenerators.ShaderGen;
+
+internal static class UnityShaderParser
```

Comment:
This actually intrigues me... I might ask you something for a completely different project.